### PR TITLE
Add initial requirements doc (placeholder)

### PR DESCRIPTION
**Pull request by jdolitsky**
~~Still need to iron out with the WG what to put in each section. Broken down into the following categories:~~

~~- MUST have~~
~~- SHOULD have~~
~~- Nice to have~~

See https://github.com/opencontainers/wg-reference-types/pull/7#issuecomment-1030339073

### CODE REVIEW

**Review by imjasonh — APPROVED** [review]

**Comment by jdolitsky** [comment]
@sudo-bmitch - I've updated the doc to simply include the raw stories from the Google Doc, with the following caveat:

"just because something is listed below does not necessarily mean this WG has reached consensus to support the use case"

I've also added a "Distilled" placeholder section which perhaps can be our next step here. Maybe from there we can get a short list of absolute requirements (maybe)

**Comment by sudo-bmitch** [comment]
@jdolitsky I'm starting from the pessimistic view and looking at how we can move forward even in the worst cases. And progress is hopefully easier as long as everyone that disagrees feels their voice was heard and vote was counted.

The new description looks good, just waiting to reconcile with my personal list to make sure nothing important was missed before adding my approval. Thanks!

File: docs/REQUIREMENTS.md
Comment (sudo-bmitch) [inline docs/REQUIREMENTS.md:65]: Channeling a BDFL that says something along the lines of "don't break user space", I think these got lost in the shuffle:

- As a user, I would like graceful degradation if the registry does not support the new ref types. This would either be some kind of limited fall back support or a clear error message from tooling. (Brandon)
- As a user, I want to be sure existing runtimes are not affected by any of the new reference types added to images. (Brandon)

Comment (jdolitsky) [reply docs/REQUIREMENTS.md:65]: my apologies I missed those! and of course, we can continue to add more of these later

**Review by sudo-bmitch — APPROVED** [review]

**Comment by nishakm** [comment]
@jdolitsky Don't you think we ought to resolve comments in the google doc before creating a PR?

**Comment by jdolitsky** [comment]
@nishakm - I guess it depends how sacred we hold the contents of this repo :)

IMO, this repo is for a "working group". If we do things right, when this work is complete, changes will land upstream and we just archive this repo, right?

The user stories in their unedited form are listed under "Member-submitted user stories (Raw)". I'm hoping that as we resolve the questions in the Google Doc, those summaries will make it into the section above it titled "Member-submitted user stories (Distilled)".

**Review by sajayantony — APPROVED** [review]
LGTM for initial merge since this is a collection of recommendations and shows progress in the discussion in the repo.

**Comment by jdolitsky** [comment]
Merged #4 and #6, but keeping this open til we've got at least 2 approvals (and general agreement)

**Comment by nishakm** [comment]
From Feb 22nd 2022 discussion, here is my list of proposed member submitted user stories:

# Top level definitions
- An OCI artifact is defined as any artifact that is to be stored in an OCI registry.
- A container image is a specialized OCI artifact as defined by the OCI [image](https://github.com/opencontainers/image-spec) and [runtime](https://github.com/opencontainers/runtime-spec) specifications.

# User stories

## Filtering
- As a user, I want to query a registry for stored artifacts by digest and tag.
- As a user, I want to query the registry for all stored artifacts that reference another artifact by its tag and digest.
- As a user, I want to query a registry for all stored artifacts of a particular type that reference another artifact by its tag and digest.
- As a user, I want to query a registry for all stored artifacts based on annotations that reference another artifact by its tag and digest.
- As a user, I want to fetch the most up-to-date artifact, collection of artifacts, or application.
- As an artifact producer, I would like to reduce the number of tags that reference artifacts.

## Backwards Compatibility
- As a user, I want to be sure that existing container runtimes are not affected by any other type of registry artifact.
- As a user, I want to move container images to and from registries that do not support reference types. 
- As an artifact producer, I want to tag artifacts that I can pull by said tag, even if they contain references to other artifacts.
- As an artifact producer, I want be sure that pushing an artifact to a repository will not affect a copy of the same artifact previously created and referenced by a manifest list existing in another repository on the same registry.
- As a tool writer, I want to identify whether a registry supports reference types or not.
- As a tool writer, I would like the option to perform a server side blob mount when copying a large artifact between repositories.

## Content Management
- As a user, I want to store one or more artifacts in a registry with a reference to another related artifact.
- As a user, when I delete an artifact, I want the option to delete one or more artifacts referencing it.
- As a user, I want to push an artifact that references another artifact that doesn't exist on the registry yet.
- As a user, I want to move an artifact and one or more artifacts referencing it from one registry to another.
- As a registry operator, I want to help users understand how they can manage the lifecycle of their artifacts.
- As a registry operator, I want to allow users to "lock" the tags to their artifacts.
- As an artifact producer, I want to update an existing artifact with a newer artifact.
- As an artifact producer, I want to push multiple artifacts concurrently (possibly from separate systems) without encountering race conditions or other errors.
- As an artifact author, I want to document other artifacts in one or more registres that my artifact requires and/or provides.
- As a user, I want assurances that the object I found really did come from the claimed supplier.
- As an artifact author, I want to add assurances that the artifact originated from me.
- As a registry operator, I want to provide users with retention policies for their artifacts.
- As a user, I want to apply timestamp or numerical ranges on my artifacts so I can apply retention policies on them. 

An open question: Is Artifact Schema a "use case"? Shouldn't we be taloring an artifact schema based on the user stories and not make artifact schema a user story?

cc: @jdolitsky @sudo-bmitch @SteveLasker @lachie83

**Comment by jdolitsky** [comment]
> An open question: Is Artifact Schema a "use case"? Shouldn't we be taloring an artifact schema based on the user stories and not make artifact schema a user story?

The schema changes would be a likely output of this working group, not a user case/user story. So yes, I agree the category should be dropped.

**Comment by jdolitsky** [comment]
@nishakm - if you want to push this into my branch, go for it. We can replace both of the "user story" sections I had drafted. Then I think we can get this merged

**Review by nishakm — APPROVED** [review]

**Comment by lachie83** [comment]
Thanks @nishakm @jdolitsky!

**Pull request merged**
